### ISSUE TITLE
Image Classification from Scratch not Training

### ISSUE THREAD

**Issue by RainehDaze** [keras-team/keras-io#286]
[Link to the relevant page.](https://keras.io/examples/vision/image_classification_from_scratch/#train-the-model) When trying to replicate this example, I ran into two problems: 

1. Not all corrupt images are deleted. After running the code, only 1578 images were deleted rather than 1590, yielding repeat warnings of the form  `Corrupt JPEG data: 2226 extraneous bytes before marker 0xd9` during training, and possibly leading to part of the following.
2. My classification accuracy stayed around 0.5 after 4 epochs, despite changing nothing in the code aside from commenting out graph visualisation, with the val_loss being nan.
```
Epoch 1/50 586/586 [==============================] - 67s 115ms/step - loss: 0.6932 - accuracy: 0.5013 - val_loss: nan - val_accuracy: 0.4957
Epoch 2/50 586/586 [==============================] - 66s 113ms/step - loss: 0.6932 - accuracy: 0.5018 - val_loss: nan - val_accuracy: 0.4957
Epoch 3/50 586/586 [==============================] - 67s 114ms/step - loss: 0.6932 - accuracy: 0.4988 - val_loss: nan - val_accuracy: 0.4957
Epoch 4/50 586/586 [==============================] - 66s 112ms/step - loss: 0.6932 - accuracy: 0.5007 - val_loss: nan - val_accuracy: 0.4957
```

At no point in the training were any errors raised or anything that might indicate that the training was proceeding incorrectly--aside, obviously, from the complete inability of the accuracy to budge.

**Comment by RainehDaze** [comment]
The accuracy and validation loss appears to be from the Windows 10 ASM issue, after comparing CPU and GPU performance on a simpler problem: 

> tensorflow/stream_executor/gpu/redzone_allocator.cc:314] Internal: Invoking GPU asm compilation is supported on Cuda non-Windows platforms only Relying on driver to perform ptx compilation. Modify $PATH to customize ptxas location.

The image corruption seems unrelated..

**Comment by sachinprasadhs** [comment]
Hi, 

I can see 1590 images are being deleted which is correct, please find the Gist here for reference https://gist.github.com/sachinprasadhs/5dd86ce3280192f28c0522a2b5572e5b.

If this is related to your environment problem, please reinstall the latest Tensorflow version using WSL-2 for windows by following steps here https://www.tensorflow.org/install/pip#windows-wsl2
